src/power_grid_model_io/data_stores/excel_file_store.py

```diff
@@ -197,6 +223,35 @@ def _check_duplicate_values(self, sheet_name: str, data: pd.DataFrame) -> Dict[i
 
         return to_rename
 
+    def _process_uuid_columns(self, data: pd.DataFrame, sheet_name: str) -> pd.DataFrame:
+        first_level = data.columns.get_level_values(0)
+        guid_columns = first_level[first_level.str.endswith("GUID")]
+
+        for guid_column in guid_columns:
+            nr = VISION_EXCEL_LAN_DICT[self._languange][DICT_KEY_NUMBER]
+            add_guid_values_to_cvtr(data, guid_column, self._uuid_cvtr)
+            new_column_name = guid_column.replace("GUID", nr)
+            if guid_column == "GUID":
+                if sheet_name in special_nodes_en:
+                    new_column_name = guid_column.replace(
+                        "GUID", VISION_EXCEL_LAN_DICT[LANGUAGE_EN][DICT_KEY_SUBNUMBER]
+                    )
+                elif sheet_name in special_nodes_nl:
+                    new_column_name = guid_column.replace(
+                        "GUID", VISION_EXCEL_LAN_DICT[LANGUAGE_NL][DICT_KEY_SUBNUMBER]
+                    )
```

Comment:
pull the mapping out of the for-loop or even in separate function.

```suggestion
        guid_columns = first_level[first_level.str.endswith("GUID")]
        sheet_key_mapping = {}
        if sheet_name in special_nodes_en:
            sheet_key_mapping = VISION_EXCEL_LAN_DICT[LANGUAGE_EN]
        elif sheet_name in special_nodes_nl:
            sheet_key_mapping = VISION_EXCEL_LAN_DICT[LANGUAGE_NL]

        for guid_column in guid_columns:
            nr = VISION_EXCEL_LAN_DICT[self._language][DICT_KEY_NUMBER]
            add_guid_values_to_cvtr(data, guid_column, self._uuid_cvtr)
            new_column_name = guid_column.replace("GUID", nr)
            if guid_column == "GUID":
                new_column_name = guid_column.replace(
                    "GUID", sheet_key_mapping[DICT_KEY_SUBNUMBER]
                )
```

Reply:
Oops, should have seen it.